### PR TITLE
Fixes #31264 - Add Github Actions support

### PR DESCRIPTION
**Pull request by upadhyeammit**
This adds support for Github Actions.

### CODE REVIEW

**Comment by jameerpathan111** [comment]
@upadhyeammit we can also add [status badge](https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/adding-a-workflow-status-badge) for github actions as follows:
`https://github.com/<OWNER>/<REPOSITORY>/workflows/<WORKFLOW_NAME>/badge.svg`
We need to add it to README.md

**Comment by upadhyeammit** [comment]
@jameerpathan111 modified to trigger actions on pr, and added badge, request to check !

File: .github/workflows/ruby_tests.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Ruby Tests
+on: pull_request
```
Comment (jameerpathan111) [inline .github/workflows/ruby_tests.yml:2]: If we want then we can run this workflow for both pull_request raised and merged.
```suggestion
on: [push, pull_request]
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
-# Foreman Maintenance [![Build Status](https://travis-ci.org/theforeman/foreman_maintain.svg?branch=master)](https://travis-ci.org/theforeman/foreman_maintain) <a href="https://codeclimate.com/github/theforeman/foreman_maintain"><img src="https://codeclimate.com/github/theforeman/foreman_maintain/badges/gpa.svg" /></a>
+# Foreman Maintenance [![Build Status](https://github.com/theforeman/foreman_maintain/workflows/Ruby%20Tests/badge.svg)](https://github.com/theforeman/foreman_maintain/workflows/Ruby%20Tests/badge.svg) <a href="https://codeclimate.com/github/theforeman/foreman_maintain"><img src="https://codeclimate.com/github/theforeman/foreman_maintain/badges/gpa.svg" /></a>
```
Comment (jameerpathan111) [inline README.md:1]:
```suggestion
# Foreman Maintain [![Build Status](https://github.com/theforeman/foreman_maintain/workflows/Ruby%20Tests/badge.svg)](https://github.com/theforeman/foreman_maintain/workflows/Ruby%20Tests/badge.svg) <a href="https://codeclimate.com/github/theforeman/foreman_maintain"><img src="https://codeclimate.com/github/theforeman/foreman_maintain/badges/gpa.svg" /></a>
```
This badge will display the status of our master branch and if we are running workflow only for only pull_request event then this won't work.
See https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/adding-a-workflow-status-badge#using-the-event-parameter

File: README.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
 
+# Foreman Maintain
```
Comment (jameerpathan111) [inline README.md:3]:
```suggestion

```

**Review by jameerpathan111 — APPROVED** [review]
LGTM

**Pull request merged**